plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -1156,6 +1160,41 @@ protected boolean isSupportedJoinCondition(ConnectorSession session, JdbcJoinCon
         return true;
     }
 
+    @Override
+    public String buildMergeRowIdConjuncts(ConnectorSession session, List<String> mergeRowIdFieldNames, List<Type> mergeRowIdFieldTypes)
+    {
+        return "ctid = ?::tid";
+    }
+
+    @Override
+    public JdbcOutputTableHandle beginDeleteTableForMerge(ConnectorSession session, JdbcTableHandle tableHandle)
```

Comment:
As I understand this is the only JDBC connector (that supports this method) so why not have `BaseJdbcClient` to be written based on Postgres?

Reply:
Postgres maybe the first connector but not the last. The connectors that support transaction insert could support MERGE that would require to implement this.

Reply:
Postgres has it's own special key `ctid`, we should avoid refer it in the `BaseJdbcClient`